### DISCUSSION TITLE
Best way to use `quacc` with a custom class containing an ASE calculator

### DISCUSSION THREAD

**Discussion by exenGT** [Quantum-Accelerators/quacc#2414] [General]
Hi @Andrew-S-Rosen,

In the past few months, I was working on developing an ASE-based customizable molecular dynamics code `d3yn` (repo currently private, has sent you a GitHub invitation link). Extensive preliminary testing with psi4 and GPAW shows some nice results, implying that it should work with any ASE-supported code. Several of our group members and I would like to use it with QChem, but the ASE QChem FileIOCalculator does not have native support for the Slurm scheduler. We have recently started looking into using the Quacc QChem calculator, but I don't know how to best integrate it with our code.

A minimal example can be found in example/claisen_rearrangement/job.py in my repo. The idea is quite simple: one initializes a `Simulator` class, which contains a `dyn` attribute (an object of a subclass of ASE Calculator). In this case, what would you think is the best way to use Quacc?

Thanks,
Jingyang

**Comment by Andrew-S-Rosen** [comment]
> Several of our group members and I would like to use it with QChem, but the ASE QChem FileIOCalculator does not have native support for the Slurm scheduler.

Can you explain what you mean by this? None of the ASE calculators have any logic for job schedulers. That is all handled by the user. As a result, all calculators should be fully compatible with any job scheduler.

>  We have recently started looking into using the Quacc QChem calculator, but I don't know how to best integrate it with our code.

Quacc might introduce a lot of dependencies you don't need, but if you are fine with that, then you can just do:

```python
from quacc.calculators.qchem import QChem
```

and use it like any other ASE calculator. It just has more features.

**Reply by exenGT** [reply]
Ah, I see! That makes sense. It appears that I was overthinking, mainly because of this line in the ASE QChem Calculator:
```
if pbs:
    self.command = 'qchem -pbs '
```
whereas no Slurm-related keyword is in there.

In any case, we would like to give the Quacc ASE Calculator a try because it supports SMD implicit solvent model. Hope it works in this case!

**Reply by Andrew-S-Rosen** [reply]
You can always modify the command to be whatever you want, so there is no restriction there.

Certainly, the quacc calculator should be much more feature-rich (it supports everything pymatgen does, including solvent models). Feel free to give that a go and use it just like any other calculator. It also runs with Custodian support by default.
